src/com/googlecode/jmxtrans/model/output/GraphiteWriter.java

```diff
@@ -101,6 +101,11 @@ public void internalWrite(Server server, Query query, ImmutableList<Result> resu
 							log.debug("Graphite Message: {}", line);
 							writer.write(line);
 							writer.flush();
```

Comment:
Don't think that will work. The `flush()` is what causes the error to emerge. `write()` just fills the datagram, but it doesn't get sent until the `flush()`. If we put the `flush()` later, our check will always show as false.